File: projects/lang-portal/backend-go/internal/models/word_review_item.go
package models

import (
	"time"
)

// WordReviewItem represents a single word review in a study session
type WordReviewItem struct {
	ID             uint         `gorm:"primaryKey" json:"id"`
	WordID         uint         `gorm:"index" json:"word_id"`
	Word           Word         `gorm:"foreignKey:WordID" json:"word,omitempty"`
	StudySessionID uint         `gorm:"index" json:"study_session_id"`
	StudySession   StudySession `gorm:"foreignKey:StudySessionID" json:"study_session,omitempty"`
	CorrectCount   int          `json:"correct_count"`
	WrongCount     int          `json:"wrong_count"`
	CreatedAt      time.Time    `json:"created_at"`
}

File: projects/capstone/backend/src/models/word_review_item.py
from sqlalchemy import Column, Integer, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from ..database import Base


class WordReviewItem(Base):
    __tablename__ = "word_review_items"

    id = Column(Integer, primary_key=True, index=True)
    word_id = Column(Integer, ForeignKey("words.id"))
    study_session_id = Column(Integer, ForeignKey("study_sessions.id"))
    correct = Column(Boolean)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    word = relationship("Word", back_populates="review_items")
    study_session = relationship("StudySession", back_populates="review_items")
